app/controllers/api/container_deployments_controller.rb

```diff
@@ -9,7 +9,12 @@ def options
       # TODO: this service is rendering resources which (a) require
       # authentication (problematic for CORS compatibility), and (b)
       # are not being properly filtered by RBAC
-      render_options(:container_deployments, ContainerDeploymentService.new.all_data)
+      if [HttpHeaders::MIQ_TOKEN, HttpHeaders::AUTH_TOKEN, "HTTP_AUTHORIZATION"].any? { |header| request.headers.include?(header) }
+        require_api_user_or_token
+        render_options(:container_deployments, ContainerDeploymentService.new.all_data)
```

Comment:
Nice @imtayadeway !!

Can we move the logic in line 12 to a method in base_controller/authentication.rb ? maybe something like request_includes_authentication.  Thanks.